lib/src/main/java/com/dynatrace/file/util/FilePollerFactory.java

```diff
@@ -0,0 +1,64 @@
+package com.dynatrace.file.util;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class FilePollerFactory {
+  private static final Logger logger = Logger.getLogger(FilePollerFactory.class.getName());
+  private static final boolean isMacOs =
+      System.getProperty("os.name", "").toLowerCase().contains("mac");
+
+  /**
+   * See {@link FilePollerFactory#getDefault(String, Duration) getDefault} for details. It calls
+   * this method with the default poll duration of 60 seconds.
+   *
+   * @param fileName The name of the file to be watched.
+   * @return An object that implements the abstract methods on {@link AbstractFilePoller}.
+   * @throws IOException if the initialization of the {@link WatchServiceBasedFilePoller} is not
+   *     successful.
+   */
+  public static AbstractFilePoller getDefault(String fileName) throws IOException {
+    return getDefault(fileName, Duration.ofSeconds(60));
+  }
+
+  /**
+   * Creates the default {@link AbstractFilePoller} based on the current OS. On Linux and Windows,
+   * the poll mechanism is based on the {@link java.nio.file.WatchService WatchService}. For macOS,
+   * where there is no native implementation for the {@link java.nio.file.WatchService WatchService}
+   * bindings, this method provides a {@link PollBasedFilePoller}, which polls the file of interest
+   * periodically (according to the {@code pollInterval}).
+   *
+   * @param fileName The name of the file to be watched.
+   * @param pollInterval The interval in which the {@link PollBasedFilePoller} polls for changes.
+   *     Only applicable on macOS.
+   * @return An object that implements the abstract methods on {@link AbstractFilePoller}.
+   * @throws IOException if the initialization of the {@link WatchServiceBasedFilePoller} is not
+   *     successful.
+   */
+  public static AbstractFilePoller getDefault(String fileName, Duration pollInterval)
+      throws IOException {
+    if (isMacOs) {
+      logger.info("Running on macOS");
```

Comment:
```suggestion
      logger.info("WatchService is not supported on macOS, using PollBasedFilePoller instead");
```

To make it a bit more clear as to why this log statement exists. :)

Reply:
I skipped this, since I the `getPollBased` method below prints that it is using a PollBasedFilePoller